packages/client-adapters/src/errors/warning-codes.js

```diff
@@ -0,0 +1,8 @@
+/** @enum {string} */
+const WarningCodes = {
+  /*
+  `hdwd` field of the response of the morphological adapter contains a number, not a string
+   */
+  NO_LEMMA_IN_HDWD: 'NO_LEMMA_IN_HDWD'
+}
```

Comment:
I think it is good to have a centralized list of error codes. This way the client will know what kinds of errors to expect from the library (this file can be imported by the client app if necessary).

Reply:
I have a question, we have two similiar classes now
`AdapterError` and `AdapterWarning`

We use errorCode from the response for `RemoteError` and errorCode for `AdapterWarning`.
ErrorCodes for RemoteErrors you defined inside `alpheios-messaging`
ErrorCodes for Warnings you defined inside `client-adapters` packages

So this way we need to know exactly the class source of the errorCode to be able to analyze it.
Because otherwises - it is only a String and not any short code.
May be it is better to think how to unify it the way you could retrieve all the data about the error from the error source? Similiar to Response errors from web server (404, 500 and so on)?

Reply:
I was having similar thoughts myself. I think we should somehow harmonize errors between different parts of the application, and think about localization of error messages. That could be a topic requiring some thought and discussion. It's probably better to create an issue for that. What do you think?

Reply:
I think it is worth to be placed to a separate issue.

Reply:
+1